src/main/kotlin/org/rust/lang/core/types/ty/TyTraitObject.kt

```diff
@@ -54,6 +55,9 @@ class TyTraitObject(
     fun getTraitBoundsTransitively(): Collection<BoundElement<RsTraitItem>> =
         traits.flatMap { it.flattenHierarchy }
 
+    val nonAutoTrait: RsTraitItem?
+        get() = traits.map { it.element }.singleOrNull { !it.isAuto }
```

Comment:
I have found several usages of `TyTraitObject#traits` where only the first item of this list is taken into account, for example `RsErrorAnnotator#checkInherentImplSameCrate` and `RsTypeDeclarationProvider#baseTypeDeclaration`. Although it is not documented that the first trait in `TyTraitObject#traits` is somehow special: it looks like the base trait is rather the one that is special there.

@mchernyavsky @vlad20012 WDYT?

Reply:
Replaced usage in `RsErrorAnnotator#checkInherentImplSameCrate` since it is related to this PR. Other usages better to address in separate PR